library/std/src/net/ip.rs

```diff
@@ -319,15 +319,7 @@ impl Ipv4Addr {
     #[stable(feature = "rust1", since = "1.0.0")]
     #[rustc_const_stable(feature = "const_ipv4", since = "1.32.0")]
     pub const fn new(a: u8, b: u8, c: u8, d: u8) -> Ipv4Addr {
-        // FIXME: should just be u32::from_be_bytes([a, b, c, d]),
-        // once that method is no longer rustc_const_unstable
-        Ipv4Addr {
-            inner: c::in_addr {
-                s_addr: u32::to_be(
-                    ((a as u32) << 24) | ((b as u32) << 16) | ((c as u32) << 8) | (d as u32),
-                ),
-            },
-        }
+        Ipv4Addr { inner: c::in_addr { s_addr: u32::from_le_bytes([a, b, c, d]) } }
```

Comment:
All of these `from_*_bytes` return a native-endian `u32`. Right now you're presenting the bytes in BE order but calling it LE -- that will sort of work on LE targets, but will get swapped on BE targets to the wrong order.

I think it should really use `from_be_bytes`, and then you still need `u32::to_be` around it.